docs/concepts/scale-up.md

```diff
@@ -0,0 +1,44 @@
+## Scaling-up the single-node to multi-node etcd cluster deployed by etcd-druid
+
+To mark the cluster for scale-up from single node to multi-node etcd just patch the etcd custom resource's `.spec.replicas` from `1` to `3`(say).
+
+### Challenges for scale-up
+1. Etcd cluster with single replica don't have any peer, so no peer communication is required hence peerUrl may or may not be TLS enabled but while scaling up from single node etcd to multi-node etcd cluster there will be a requirement to have a peer communication between each members of etcd cluster. Peer communication is required for various reasons like to get in a sync with each other cluster state, data and to perform leader election or any cluster wide operation like to remove/add a member etc. Hence in multi-node etcd cluster we need to have TLS enable peer url for peer communication.
+2. Providing the correct configuration to start a new etcd members as it is different from boostrapping a cluster since this new etcd members will join the existing cluster.
+
+### Approach
+We first went through the etcd doc of [update-advertise-peer-urls](https://etcd.io/docs/v3.4/op-guide/runtime-configuration/#update-advertise-peer-urls) to find out information regarding to update peerUrl. Interestingly etcd doc have mentioned this:
+```
+To update the advertise peer URLs of a member, first update it explicitly via member command and then restart the member.
+```
+
+But we can't assume peerUrl is not TLS enabled for single-node cluster as it depends on end-user, user may or may not enable the TLS for peerUrl for single node etcd cluster. so, How to detect whether peerUrl was enabled or not when cluster is marked for scale-up ?
+
+### Detection of peerUrl TLS enabled or not
+For this we have used member lease annotation `member.etcd.gardener.cloud/tls-enabled` set by backup-restore sidecar of etcd. As etcd configuration is provided by backup-restore, so backup-restore can find out whether TLS is enabled or not and accordingly set this annotation `member.etcd.gardener.cloud/tls-enabled: "true"` or `member.etcd.gardener.cloud/tls-enabled: "false"` in member lease object.
```

Comment:
```suggestion
For this we use an annotation in the member lease `member.etcd.gardener.cloud/tls-enabled` set by backup-restore sidecar of etcd. As etcd configuration is provided by backup-restore, it can find out whether TLS is enabled or not and accordingly from the configuration and set this annotation `member.etcd.gardener.cloud/tls-enabled:` to either `true` or `false` in member lease object.
```